worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go

```diff
@@ -737,25 +737,41 @@ func buildBenthosGenerateSourceConfigResponses(
 			count = tableOpt.Count
 		}
 
-		mapping, err := buildMutationConfigs(ctx, transformerclient, tableMapping.Mappings, columnInfo)
+		jsCode, err := extractJsFunctionsAndOutputs(ctx, transformerclient, tableMapping.Mappings)
 		if err != nil {
 			return nil, err
 		}
 
+		mutations, err := buildMutationConfigs(ctx, transformerclient, tableMapping.Mappings, columnInfo)
+		if err != nil {
+			return nil, err
+		}
+
+		// for the generate input, benthos requires a mapping, so falling back to a
+		// generic empty object if the mutations are empty
+		if mutations == "" {
+			mutations = "root = {}"
```

Comment:
I think this makes sense here to accomodate the case where no generators have been supplied save for javascript ones.